utils/types/types.go

```diff
@@ -44,11 +56,21 @@ type ConfigEnvI interface {
 	ReplaceConfigWithEnvVariables(workspaceConfig []byte) (updatedConfig []byte)
 }
 
-// ReportingI is interface to report metrics
-type ReportingI interface {
+// Reporting is interface to report metrics
+type Reporting interface {
 	WaitForSetup(ctx context.Context, clientName string) error
-	AddClient(ctx context.Context, c Config)
 	Report(metrics []*PUReportedMetric, txn *sql.Tx)
+	AddClient(ctx context.Context, c Config)
+}
+
+// FacadeInterface for reporting
+type MasterReporting interface {
+	Reporting
+	GetReportingInstance(reporterType ReporterType) Reporter
+}
+
+type Reporter interface {
+	Reporting
 	IsPIIReportingDisabled(string) bool
 }
```

Comment:
Not sure if this interface is needed, or at least if we need to define here and export it.

Reply:
>Not sure if this interface is needed

This is an interface for reporting instances(error_reporting or reporting). I think it'd be better if we have this

>if we need to define here and export it.

As far as defining it is concerned. I was thinking moving this interface to mediator.go or setup.go. WDYT ? Any other place you feel it is better suited, please let me know